# map D3D10 texture1D subresource zero for every mip and array slice
--- a/src/d3d10/d3d10_texture.cpp
+++ b/src/d3d10/d3d10_texture.cpp
@@ -5,7 +5,7 @@
 HRESULT STDMETHODCALLTYPE
 MTLD3D10Texture1D::Map(UINT Subresource, D3D10_MAP MapType, UINT MapFlags, void **ppData) {
   D3D11_MAPPED_SUBRESOURCE subresource{};
-  HRESULT hr = d3d11_context->Map(d3d11, Subresource, D3D11_MAP(MapType), MapFlags, &subresource);
+  HRESULT hr = d3d11_context->Map(d3d11, 0, D3D11_MAP(MapType), MapFlags, &subresource);
   if (hr == S_OK)
     *ppData = subresource.pData;
   return hr;
